app/Platform/Actions/UpdateGameMetricsAction.php

```diff
@@ -4,47 +4,106 @@
+            $achievementsPublishedChange = $game->achievements_published - $game->getOriginal('achievements_published');
+            $pointsTotalChange = $game->points_total - $game->getOriginal('points_total');
+            $playersTotalChange = $game->players_total - $game->getOriginal('players_total');
+            $playersHardcoreChange = $game->players_hardcore - $game->getOriginal('players_hardcore');
+        }
+        $pointsWeightedBeforeUpdate = $game->TotalTruePoints;
+
+        $game->save();
+
+        // dispatch(new UpdateGameAchievementsMetricsJob($game->id))->onQueue('game-metrics');
+        app()->make(UpdateGameAchievementsMetricsAction::class)
+            ->execute($game);
+        $game->refresh();
+        $pointsWeightedChange = $game->TotalTruePoints - $pointsWeightedBeforeUpdate;
+
+        GameMetricsUpdated::dispatch($game);
 
-        /*
-         * update unpublished achievements metrics
-         */
-        // $this->attributes['achievements_unpublished'] = $this->achievements()->unpublished()->count();
+        // TODO dispatch events for achievement set and game metrics changes
+        $tmp = $achievementsPublishedChange;
+        $tmp = $pointsTotalChange;
+        $tmp = $pointsWeightedChange;
+        $tmp = $playersTotalChange;
+        $tmp = $playersHardcoreChange;
 
-        // $this->attributes['players_total'] = $this->players()->count();
+        // ad-hoc updates for player games, so they can be updated the next time a player updates their profile
+        // Note that those might be multiple thousand entries depending on a game's players count
 
-        // $game->save();
+        $updateReason = null;
+        if ($pointsWeightedChange) {
+            $updateReason = 'weighted_points_outdated';
+        }
+        if ($achievementSetVersionChanged) {
+            $updateReason = 'version_mismatch';
+        }
+        if ($updateReason) {
+            $game->playerGames()
+                ->where(function ($query) use ($game, $updateReason) {
+                    if ($updateReason === 'weighted_points_outdated') {
+                        $query->whereNot('points_weighted_total', '=', $game->TotalTruePoints)
+                            ->orWhereNull('points_weighted_total');
+                    }
+                    if ($updateReason === 'version_mismatch') {
+                        $query->whereNot('achievement_set_version_hash', '=', $game->achievement_set_version_hash)
+                            ->orWhereNull('achievement_set_version_hash');
+                    }
+                })
+                ->update([
+                    'update_status' => $updateReason,
+                    'achievements_total' => $game->achievements_published,
+                    'points_total' => $game->points_total,
+                    'points_weighted_total' => $game->TotalTruePoints,
+                ]);
+        }
```

Comment:
See https://github.com/RetroAchievements/RAWeb/pull/1874#discussion_r1335158595
> How the update of player games based on achievement set version changes will work in detail is still open - most likely it will be a scheduled task that takes care of scheduling all the player games marked for update (update_status) or triggered by a user logging in.

I'm still trying to find a good and scalable way to cascade into player games metrics update which also will make sure that badge eligibility is taken into account.

> // ad-hoc updates for player games, so they can be updated the next time a player updates their profile
> // Note that those might be multiple thousand entries depending on a game's players count